angular/projects/admin-nrpti/src/app/mines/mines-add-edit/mines-add-edit.component.ts

```diff
@@ -164,11 +184,58 @@ export class MinesAddEditComponent implements OnInit, OnDestroy {
    * @memberof MinesAddEditComponent
    */
   togglePublish(event) {
-    this.myForm.controls.publish.setValue(event.checked);
+    if (!event.checked) {
+      // always allow unpublishing
+      this.myForm.controls.publish.setValue(event.checked);
+    } else if (this.canPublish) {
+      // conditionally allow publishing
+      this.myForm.controls.publish.setValue(event.checked);
+    }
 
     this._changeDetectionRef.detectChanges();
   }
 
+  /**
+   * Return true if the record meets the criteria to be published, false otherwise.
+   *
+   * @returns {boolean}
+   * @memberof MinesAddEditComponent
+   */
+  checkCanPublish(): boolean {
+    return (
+      this.mine &&
+      this.myForm &&
+      this.myForm.get('description').value &&
+      this.myForm.get('summary').value &&
+      this.myForm.get('type').value &&
+      this.mine.name &&
+      this.mine.status &&
+      this.mine.permitNumbers &&
+      this.mine.permitNumbers.length &&
+      this.mine.tailingsImpoundments &&
```

Comment:
This one is still going to fail if it is 0.

Reply:
Ohh right, sorry I misunderstood what you meant.  Will adjust

Reply:
Updated in latest commit.

I also noticed that our mine.ts was defaulting tailingsImpoundments to null, so I updated that to be 0 instead.